source/extensions/filters/http/jwt_authn/matcher.cc

```diff
@@ -188,6 +189,41 @@ class PathSeparatedPrefixMatcherImpl : public BaseMatcherImpl {
 };
 } // namespace
 
+class PathMatchPolicyMatcherImpl : public BaseMatcherImpl {
+public:
+  PathMatchPolicyMatcherImpl(const RequirementRule& rule,
+                             Server::Configuration::CommonFactoryContext& context)
+      : BaseMatcherImpl(rule, context) {
+    auto& factory = Config::Utility::getAndCheckFactory<Router::PathMatcherFactory>(
+        rule.match().path_match_policy());
+    ProtobufTypes::MessagePtr config = Envoy::Config::Utility::translateAnyToFactoryConfig(
+        rule.match().path_match_policy().typed_config(),
+        ProtobufMessage::getStrictValidationVisitor(), factory);
+
+    absl::StatusOr<Router::PathMatcherSharedPtr> matcher = factory.createPathMatcher(*config);
+
+    if (!matcher.ok()) {
+      throw EnvoyException(std::string(matcher.status().message()));
+    }
+
+    uri_template_matcher_ = matcher.value();
+  }
+
+  bool matches(const Http::RequestHeaderMap& headers) const override {
+    if (BaseMatcherImpl::matchRoute(headers) &&
+        uri_template_matcher_->match(headers.getPathValue())) {
+      ENVOY_LOG(debug, "Path match policy requirement '{}' matched.",
+                uri_template_matcher_->uriTemplate());
+      return true;
+    }
+
+    return false;
+  }
+
+private:
+  Router::PathMatcherSharedPtr uri_template_matcher_;
```

Comment:
To improve const correctness for the state that is shared across multiple threads, please make this `const`
```suggestion
  const Router::PathMatcherSharedPtr uri_template_matcher_;
```

You will then need to create a function for to initialize uri_template_matcher_ so it can be done in the constructor initializer list.

Reply:
done